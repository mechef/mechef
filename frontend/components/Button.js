// @flow

import * as React from 'react';

import * as styles from '../utils/styleVariables';

type Props = {
  buttonStyle?: 'primary' | 'greenBorderOnly' | 'whiteBorderOnly',
  size?: 'small' | 'medium' | 'expanded',
  children?: React.Node,
  onClick: () => mixed,
};

const Button = (props: Props) => (
  <button
    type="button"
    className={`
      button
      ${props.buttonStyle ? props.buttonStyle : ''}
      ${props.size ? props.size : ''}
    `}
    onClick={props.onClick}
  >
    {props.children}
    <style jsx>
      {`
        .button {
          outline: none;
          border: 0;
          border-radius: ${styles.borderRadius};
          height: ${styles.btnHeight};
          cursor: pointer;
          font-weight: ${styles.fontWeight};
          font-size: ${styles.fontSize};
          line-height: ${styles.lineHeight};
        }
        .primary {
          background-color: ${styles.primaryColor};
          color: ${styles.btnTextColor};
        }
        .primary:hover {
          background-color: ${styles.primaryBtnHoverColor};
        }
        .greenBorderOnly {
          background-color: ${styles.transparent};
          border: solid 1px ${styles.primaryColor};
          color: ${styles.primaryColor};
        }
        .greenBorderOnly:hover {
          background-color: ${styles.primaryColor};
          color: ${styles.btnTextColor};
        }
        .whiteBorderOnly {
          background-color: ${styles.transparent};
          border: solid 1px ${styles.btnTextColor};
          color: ${styles.btnTextColor};
        }
        .whiteBorderOnly:hover {
          background-color: ${styles.btnTextColor};
          color: rgba(0, 0, 0, 0.5);
        }
        .small {
          width: ${styles.smallBtnWidth};
        }
        .medium {
          width: ${styles.mediumBtnWidth};
        }
        .expanded {
          width: 100%;
        }
        @media screen and (max-width: 768px) {
          .small {
            width: 90%;
          }
          .medium {
            width: 90%;
          }
        }
        @media screen and (max-width: 540px) {
          .small {
            width: 100%;
          }
        }
      `}
    </style>
  </button>
);

Button.defaultProps = {
  buttonStyle: 'primary',
  size: 'small',
  children: '',
  onClick: () => {},
};

export default Button;
