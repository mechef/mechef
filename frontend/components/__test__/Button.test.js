import renderer from 'react-test-renderer';
import React from 'react';

import Button from '../Button';


describe('Pages With Snapshot Testing', () => {
  it('snapshot Button Component', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
