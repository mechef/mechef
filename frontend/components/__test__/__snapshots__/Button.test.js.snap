// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot Button Component 1`] = `
<button
  className="jsx-3459932710 
      button
      primary
      small
    "
  onClick={[Function]}
  type="button"
>
  
</button>
`;
